api/secrets.proto

```diff
@@ -0,0 +1,136 @@
+	// - Returns `NotFound` if the a secret named `RemoveSecretRequest.ID` is not found.
+	// - Returns an error if the deletion fails.
+	rpc RemoveSecret(RemoveSecretRequest) returns (RemoveSecretResponse) {
+		option (docker.protobuf.plugin.tls_authorization) = { roles: "swarm-manager" };
+	}
+}
+
+// GetSecretRequest is the request to get a `Secret` object given a secret id.
+message GetSecretRequest {
+	string secret_id = 1 [(gogoproto.customname) = "SecretID"];
+}
+
+// GetSecretResponse contains the Secret corresponding to the id in
+// `GetSecretRequest`, but the `SecretValue.Spec.Data` field in each `SecretValue`
+// object in the `Secret` should be nil instead of actually containing
+// the secret bytes.
+message GetSecretResponse {
+	Secret secret = 1;
+}
+
+// ListSecretRequest is the request to list all secrets in the secret store,
+// or all secrets filtered by (id or id prefix or name or name prefix).
+message ListSecretsRequest {
+	message Filters {
+		repeated string secret_ids = 1 [(gogoproto.customname) = "SecretIDs"];
+		repeated string id_prefixes = 2 [(gogoproto.customname) = "IDPrefixes"];
+		repeated string names = 3;
+		repeated string name_prefixes = 4;
+	}
+
+	Filters filters = 1;
+}
+
+// ListSecretResponse contains a list of all the secrets that match the name or
+// name prefix filters provided in `ListSecretRequest`.  The `SecretValue.Spec.Data`
+// field in each `SecretValue` object in each `Secret` should be nil instead of
+// actually containing the secret bytes.
+message ListSecretsResponse {
+	repeated Secret secrets = 1;
+}
+
+// CreateSecretRequest specifies a new secret (it will not update an existing
+// secret) to create.
+message CreateSecretRequest {
+	SecretValueSpec spec = 1;
+}
+
+// CreateSecretResponse contains the newly created `Secret`` corresponding to the
+// name in `CreateSecretRequest`.  Since it's a new secret, there should just be
+// the one version, and hence only one `SecretValue` object in the `Secret`.
+// The `SecretValue.Spec.Data` field should be nil instead of actually
+// containing the secret bytes.
+message CreateSecretResponse {
+	Secret secret = 1;
+}
+
+// UpdateSecretRequest specifies a new secret version to add to an existing secret
+// of the same name.
+message UpdateSecretRequest {
+	SecretValueSpec spec = 1;
+}
+
+// UpdateSecretResponse contains the Secret corresponding to the name and type of
+// the spec provided in `UpdateSecretRequest`, and it should be updated with an
+// additional `SecretValue` object containing said spec.  The
+// `SecretValue.Spec.Data` field in each `SecretValue` object in the `Secret`
+// object should be nil instead of actually containing the secret bytes.
+message UpdateSecretResponse {
+	Secret secret = 1;
+}
+
+// RemoveSecretRequest contains the ID of the secret that should be removed.  This
+// removes all versions of the secret.
+message RemoveSecretRequest {
+	string secret_id = 1 [(gogoproto.customname) = "SecretID"];
```

Comment:
I saw the conversation above, but is there a way to remove a specific revision of a secret?